src/chains/ethereum/ethereum/tests/transaction-pool.test.ts

```diff
@@ -95,421 +96,714 @@ describe("transaction pool", async () => {
+        futureNonceTx.serialized.toString()
+      );
+      const futureInOrigin = findIn(futureNonceTx.hash.toBuffer(), origins);
+      assert.strictEqual(futureInOrigin, undefined);
+    });
+
+    it("can be cleared/emptied", async () => {
+      const txPool = new TransactionPool(options.miner, blockchain, origins);
+      const transaction = TransactionFactory.fromRpc(rpcTx, common);
+
+      await txPool.prepareTransaction(transaction, secretKey);
+      const { pending } = txPool.executables;
+      // our executable transaction should be found in the pending queue
+      const found = findIn(transaction.hash.toBuffer(), pending);
+      assert.strictEqual(
+        found.serialized.toString(),
+        transaction.serialized.toString()
+      );
+
+      const futureNonceTx = TransactionFactory.fromRpc(futureNonceRpc, common);
+      const futureIsExecutable = await txPool.prepareTransaction(
+        futureNonceTx,
+        secretKey
+      );
+      assert(!futureIsExecutable); // our transaction is not executable
+      // our non executable transaction should be found in the origins queue
+      const foundInOrigins = findIn(futureNonceTx.hash.toBuffer(), origins);
+      assert.strictEqual(
+        foundInOrigins.serialized.toString(),
+        futureNonceTx.serialized.toString()
+      );
+
+      txPool.clear();
+      // both queues should be empty
+      assert.strictEqual(pending.values.length, 0);
+      assert.strictEqual(origins.values.length, 0);
+    });
+
+    it("emits an event when a transaction is ready to be mined", async () => {
+      const txPool = new TransactionPool(options.miner, blockchain, origins);
+      const transaction = TransactionFactory.fromRpc(rpcTx, common);
+
+      await txPool.prepareTransaction(transaction, secretKey);
+      const drainPromise = txPool.once("drain");
+      txPool.drain();
+      await drainPromise;
+
+      const { pending } = txPool.executables;
+      // our executable transaction should be found in the pending queue after the drain event
+      const found = findIn(transaction.hash.toBuffer(), pending);
+      assert.strictEqual(
+        found.serialized.toString(),
+        transaction.serialized.toString()
+      );
+    });
   });
 
-  it("replaces future nonce transactions in the future queue", async () => {
-    const txPool = new TransactionPool(options.miner, blockchain, origins);
-    const futureNonceTx = TransactionFactory.fromRpc(futureNonceRpc, common);
-    const isExecutable = await txPool.prepareTransaction(
-      futureNonceTx,
-      secretKey
-    );
-    assert(!isExecutable); // our transaction is not executable
-    // our non executable transaction should be found in the origins queue
-    const found = findIn(futureNonceTx.hash.toBuffer(), origins);
-    assert.strictEqual(
-      found.serialized.toString(),
-      futureNonceTx.serialized.toString()
-    );
+  describe("nonce generation and validation", async () => {
```

Comment:
This is the new describe block. I don't know what github is doing here with this diff, I'd just look at this new describe block in your editor instead of here.